apps/meteor/packages/linkedin-oauth/linkedin-server.js

```diff
@@ -53,33 +14,38 @@ const getTokenResponse = async function (query) {
 	let responseContent;
 	try {
 		// Request an access token
-		const request = await fetch('https://api.linkedin.com/uas/oauth2/accessToken', {
+		const body = new URLSearchParams({
+			grant_type: 'authorization_code',
+			client_id: config.clientId,
+			client_secret: OAuth.openSecret(config.secret),
+			code: query.code,
+			redirect_uri: OAuth._redirectUri('linkedin', config),
+		});
+
+		const response = await fetch('https://api.linkedin.com/uas/oauth2/accessToken', {
 			method: 'POST',
-			body: JSON.stringify({
-				grant_type: 'authorization_code',
-				client_id: config.clientId,
-				client_secret: OAuth.openSecret(config.secret),
-				code: query.code,
-				redirect_uri: OAuth._redirectUri('linkedin', config),
-			}),
+			headers: {
+				'Content-Type': 'application/x-www-form-urlencoded',
+			},
+			body,
 		});
-		responseContent = await request.text();
+
+		responseContent = await response.json();
+
+		if (!response.ok) {
+			throw new Error(responseContent.error_description);
+		}
+
```

Comment:
```suggestion
		if (!response.ok) {
			throw new Error(responseContent.error_description);
		}

		responseContent = await response.json();
```

Check if it was sucessful before trying to read the json.